api/src/certification/configuration/domain/usecases/import-sco-whitelist.js

```diff
@@ -12,12 +12,30 @@ export const importScoWhitelist = withTransaction(
    */
   async ({ externalIds = [], centerRepository }) => {
     await centerRepository.resetWhitelist();
-    const numberOfUpdatedLines = await centerRepository.addToWhitelistByExternalIds({ externalIds });
+    const updatedExternalIds = await centerRepository.addToWhitelistByExternalIds({ externalIds });
+
+    const uniqueExternalIds = [...new Set(externalIds)];
+    const externalIdsOnError = uniqueExternalIds.filter((externalId) => {
+      return !updatedExternalIds.includes(externalId);
+    });
+
+    if (externalIdsOnError.length > 0) {
+      const indexShiftAndCsvHeader = 2;
```

Comment:
Proposition : pour alleger le corps de la fonction principale (cas nominal) je propose d'extraire toute la generation du rapport d'erreur dans une fonction


```suggestion
      throw new InvalidScoWhitelistError({
        lineNumbersWithError : _getErrorReport(externalIds, externalIdsOnError),
      });
```

En veritie pour pourrait aller jusque la ligne `uniqueExternalIds` et la fonction principale pourrait ressembler ainsi a 

```js

async ({ externalIds = [], centerRepository }) => {
    await centerRepository.resetWhitelist();
    const updatedExternalIds = await centerRepository.addToWhitelistByExternalIds({ externalIds });

    const errorReport = _getErrorReport(updatedExternalIds);
    if (errorReport.length > 0) {
      throw new InvalidScoWhitelistError({lineNumbersWithError});
    }
  },

const _getErrorReport = ()=>{ /* ....... */ };


```